openbas-api/src/main/java/io/openbas/executors/caldera/service/CalderaExecutorService.java

```diff
@@ -162,63 +154,76 @@ public void run() {
-    matchingExistingEndpoint.setLastSeen(external.getLastSeen());
-    matchingExistingEndpoint.setExternalReference(external.getExternalReference());
-    matchingExistingEndpoint.setName(external.getName());
-    matchingExistingEndpoint.setIps(external.getIps());
-    matchingExistingEndpoint.setHostname(external.getHostname());
-    matchingExistingEndpoint.setProcessName(external.getProcessName());
-    matchingExistingEndpoint.setPlatform(external.getPlatform());
-    matchingExistingEndpoint.setArch(external.getArch());
-    matchingExistingEndpoint.setExecutor(this.executor);
-    if ((now().toEpochMilli() - matchingExistingEndpoint.getClearedAt().toEpochMilli())
-        > CLEAR_TTL) {
+      @NotNull final Agent agent, @NotNull final Endpoint existingEndpoint) {
+    existingEndpoint.setLastSeen(toInstant(agent.getLast_seen()));
+    existingEndpoint.setExternalReference(agent.getPaw());
+    existingEndpoint.setName(agent.getHost());
+    existingEndpoint.setIps(agent.getHost_ip_addrs());
+    existingEndpoint.setHostname(agent.getHost());
+    existingEndpoint.setProcessName(agent.getExe_name());
+    existingEndpoint.setPlatform(toPlatform(agent.getPlatform()));
+    existingEndpoint.setArch(toArch(agent.getArchitecture()));
+    existingEndpoint.setExecutor(this.executor);
+    if ((now().toEpochMilli() - existingEndpoint.getClearedAt().toEpochMilli()) > CLEAR_TTL) {
       try {
-        log.info("Clearing endpoint " + matchingExistingEndpoint.getHostname());
+        log.info("Clearing endpoint " + existingEndpoint.getHostname());
         Iterable<Injector> injectors = injectorService.injectors();
         injectors.forEach(
             injector -> {
               if (injector.getExecutorClearCommands() != null) {
-                this.calderaExecutorContextService.launchExecutorClear(
-                    injector, matchingExistingEndpoint);
+                this.calderaExecutorContextService.launchExecutorClear(injector, existingEndpoint);
               }
             });
-        matchingExistingEndpoint.setClearedAt(now());
+        existingEndpoint.setClearedAt(now());
       } catch (RuntimeException e) {
         log.info("Failed clear agents");
       }
     }
-    this.endpointService.updateEndpoint(matchingExistingEndpoint);
+    this.endpointService.updateEndpoint(existingEndpoint);
   }
 
-  private Instant toInstant(@NotNull final String lastSeen) {
+  @VisibleForTesting
+  protected Instant toInstant(@NotNull final String lastSeen) {
     String pattern = "yyyy-MM-dd'T'HH:mm:ss'Z'";
     DateTimeFormatter dateTimeFormatter = DateTimeFormatter.ofPattern(pattern, Locale.getDefault());
     LocalDateTime localDateTime = LocalDateTime.parse(lastSeen, dateTimeFormatter);
     ZonedDateTime zonedDateTime = localDateTime.atZone(UTC);
     return zonedDateTime.toInstant();
   }
+
+  public void setExecutor(Executor executor) {
+    this.executor = executor;
+  }
```

Comment:
**question**:
Why do you need this ?

Reply:
for testing, because I was not using a mock I was not able to inject it. I can update the access modifier to protected as it won't be used anywhere else.